File: Java-Versions/Question1/src/Question1.java

/*
    Java
    Question1
    Adam Giaccaglia

 */

public class Question1 {
    public static void main( String[] args ) {

        Animal[] animals = new Animal[4];

        animals[0] = new Animal();
        animals[1] = new Cow();
        animals[2] = new Pig();
        animals[3] = new Snake();

        for (int i = 0; i < animals.length; i++)
        {
            animals[i].talk();
        }

    }
}

class Animal
{
    private int Leg;
    public Animal()
    {
        Leg = 4;
    }

    public void talk()
    {
        System.out.println("Animals Can't Talk!");
    }
}

class Cow extends Animal {
    // functions are virtual by default
    public void talk() {
      System.out.println("Moo!");
  }
}

class Pig extends Animal {
    public void talk() {
        System.out.println("Grunt!");
    }
}
class Snake extends Animal {

}

File: C PLusPlus Versions/C PLusPlus Versions/Question1.cpp
// C PLusPlus Versions.cpp : Defines the entry point for the console application.
//


/*

	Question 1 
	C++
	Adam Giaccaglia

*/
#include "stdafx.h"
#include <iostream>
#include <string>

using namespace std;


class Animal
{
private:
	int Leg;

public:
	Animal()
	{
		Leg = 4;
	}
	virtual void talk()
	{
		cout << ("Animals Can't Talk!");
	}
};

class Cow : public Animal
{
public:
	void talk()
	{
		cout << ("Moo!");
	}

};

class Pig : public Animal
{
public:
	void talk()
	{
		cout << ("Grunt!");
	}

};

//public to override default private inheritance
class Snake : public Animal
{
public:

};

int _tmain(int argc, _TCHAR* argv[])
{
	//pointer for dyanmic binding and allocation
	Animal* animals[4];

	animals[0] = new Animal();
	animals[1] = new Cow();
	animals[2] = new Pig();
	animals[3] = new Snake();


	for (int i = 0; i < 4; i++)
	{
		animals[i]->talk();
		cout << endl;
		delete animals[i];
	}
	
	cin.get();
}

